lib/coap/connection.ex

```diff
@@ -213,18 +274,34 @@ defmodule CoAP.Connection do
   end
 
   # send message to peer from client
+  # The client should not send a message id, that is managed by the connection
   defp deliver_message(
-         %Message{type: type} = message,
-         %{phase: :idle, next_message_id: message_id} = state
+         %Message{type: type, message_id: message_id} = message,
+         %{phase: :idle, next_message_id: next_message_id} = state
        ) do
-    %{message | message_id: message_id}
+    {data, block, payload} = Payload.next_segment(message.payload, @default_payload_size)
```

Comment:
This splits a message when it needs to, or leaves it whole when it doesn't. Later we'll respect control size.